pkg/registries/ecr/transport.go

```diff
@@ -46,11 +47,12 @@ func (t *awsTransport) RoundTrip(req *http.Request) (*http.Response, error) {
 	)
 }
 
-func (t *awsTransport) isExpiredNoLock() bool {
-	return t.expiresAt == nil || t.expiresAt.After(time.Now())
+func (t *awsTransport) isValidNoLock() bool {
+	return t.expiresAt != nil && t.expiresAt.After(time.Now())
```

Comment:
could you also just flip `After` to `Before`? Makes sense to flip the whole function for consistence though

Reply:
oh I meant instead of flipping  `==` to `!=` and `||` to `&&`

Reply:
Now I'm confused. Could you post a suggestion 😄 ? The reason I flipped it in the first place is that golang `oauth2.Token` also uses `Valid` and not `IsExpired`, see https://pkg.go.dev/golang.org/x/oauth2#Token.Valid. Of course one could also flip `After` -> `Before` in the original `isExpiredNoLock`, that would also fix it but I think it's slightly more difficult to reason about.

Reply:
I was saying instead of making all these changes you could have just done

`t.expiresAt.After(time.Now()) -> t.expiresAt.Before(time.Now())`

However, the changes you made, `t.expiresAt != nil && t.expiresAt.After(time.Now())` and switching it to `isValidNoLock` makes sense since it mirrors what we do for Google